apps/pgsql/patroni/openshift/deployment.yaml

```diff
@@ -196,16 +230,41 @@ objects:
             value: 0.0.0.0:5432
           - name: PATRONI_RESTAPI_LISTEN
             value: 0.0.0.0:8008
-          image: ${IMAGE_REGISTRY}/${IMAGE_STREAM_NAMESPACE}/${IMAGE_STREAM_TAG}
-          # Because we are using image reference to a tag, we need to always pull the image otherwise
-          #   we end up with outdated/out-of-sync image depending on the node where it is running
-          imagePullPolicy: Always
-          name: postgresql
+          lifecycle:
+            preStop:
+              exec:
+                command:
+                  - /usr/bin/env
+                  - bash
+                  - -c
+                  - |
+                    # switch leader pod if the current pod is the leader
+                    if curl --fail http://localhost:8008/read-write; then
+                      patronictl switchover --force
+                    fi
+          livenessProbe:
```

Comment:
can you change the livenessprobe to use the same script as readinessProbe?

Reply:
It doesn't make much sense to set both to run the same script (you'd just be doing it twice on the pod every 10 seconds). Liveness checks whether the container is running; Readiness checks if the container is ready to accept (postgres) network traffic. There are valid states/modes where you can be live, but not ready (pod would be marked as Unready, and service would not direct net traffic to it). Setting both liveness and readiness to the same thing defeats the purpose of this subtle nuance.

The way it's written right now, Liveness will check if Patroni (and API) is up and running. Readiness will check disk, postgres, and the _status_ of this Patroni instance. There will be situations where you may want to have Patroni running, but postgres itself is not (i.e. manual intervention by running `patronictl pause`). If you have both doing the readiness check, you will end up killing the containers repeatedly and be unable to proceed with potential recovery options.

Reference this incident: https://github.com/bcgov/nr-get-token/wiki/Patroni-Troubleshooting#total-cluster-failure-recovery